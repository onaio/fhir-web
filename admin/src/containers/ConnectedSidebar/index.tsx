import { getUser, isAuthenticated, getExtraData } from '@onaio/session-reducer';
import { connect } from 'react-redux';
import { Store } from 'redux';
import Sidebar from '../../components/page/Sidebar';

/** Connect the component to the store */

/** map state to props */
const mapStateToProps = (state: Partial<Store>) => {
  const result = {
    authenticated: isAuthenticated(state),
    user: getUser(state),
    extraData: getExtraData(state),
  };
  return result;
};

const mapDispatchToProps = {};

/** create connected component */

/** Connected Sidebar component
 */
const ConnectedSidebar = connect(mapStateToProps, mapDispatchToProps)(Sidebar);

export default ConnectedSidebar;
