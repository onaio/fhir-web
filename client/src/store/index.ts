// going to use this store initializer instead of the one in
// @opensrp/store package, for some reason its not working
// wehn passed to the redux provider
import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import {
  reducerName as keycloakUsersReducerName,
  reducer as keycloakUsersReducer,
} from '@opensrp/store';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;
// added this segment because registering from the user management
// package component is failing
defaultReducers[keycloakUsersReducerName] = keycloakUsersReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});

export default store;
