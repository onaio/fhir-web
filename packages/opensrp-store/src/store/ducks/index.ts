/** users ducks modules: actions, actionCreators, reducer and selectors */
import { Dictionary } from '@onaio/utils';
import { keyBy, values } from 'lodash';
import intersect from 'fast_array_intersect';
import { AnyAction, Store } from 'redux';
import { createSelector, OutputParametricSelector } from 'reselect';
import SeamlessImmutable from 'seamless-immutable';

/** The reducer name */
export const reducerName = 'keycloakUsers';

/** Interface for user json object */
export interface KeycloakUser {
  access?: {
    manageGroupMembership: boolean;
    view: boolean;
    mapRoles: boolean;
    impersonate: boolean;
    manage: boolean;
  };
  createdTimestamp?: number;
  disableableCredentialTypes?: string[];
  email?: string;
  emailVerified?: boolean;
  enabled?: boolean;
  firstName: string;
  id: string;
  lastName: string;
  notBefore?: number;
  requiredActions?: string[];
  totp?: boolean;
  username: string;
}

// actions

/** action type for fetching keycloak users */
export const KEYCLOAK_USERS_FETCHED = 'keycloak/reducer/users/userS_FETCHED';
/** action type for removing keycloak users */
export const REMOVE_KEYCLOAK_USERS = 'keycloak/reducer/users/REMOVE_KEYCLOAK_USERS';

/** interface action to add users to store */
export interface FetchKeycloakUsersAction extends AnyAction {
  usersById: Dictionary<KeycloakUser>;
  type: typeof KEYCLOAK_USERS_FETCHED;
}

/** Interface for removeusersAction */
export interface RemoveKeycloakUsersAction extends AnyAction {
  usersById: Dictionary<KeycloakUser>;
  type: typeof REMOVE_KEYCLOAK_USERS;
}

/** Create type for users reducer actions */
export type KeycloakUsersActionTypes =
  | FetchKeycloakUsersAction
  | RemoveKeycloakUsersAction
  | AnyAction;

// action Creators

/**
 * Fetch users action creator
 *
 * @param {Array} usersList - keycloak users
 * @returns {object} - the dispatcher to remove the user
 */
export const fetchKeycloakUsers = (usersList: KeycloakUser[] = []): FetchKeycloakUsersAction => {
  return {
    usersById: keyBy(usersList, (user: KeycloakUser) => user.id),
    type: KEYCLOAK_USERS_FETCHED,
  };
};

/** Remove users action creator
 *
 * @returns {object} - the dispatcher to remove the user
 */
export const removeKeycloakUsers = (): RemoveKeycloakUsersAction => {
  return {
    usersById: {},
    type: REMOVE_KEYCLOAK_USERS,
  };
};

// The reducer

/** interface for keycloak users state in redux store */
export interface KeycloakUsersState {
  usersById: Dictionary<KeycloakUser> | Dictionary;
}

/** Create an immutable keycloak users state */
export type ImmutableKeycloakUsersState = KeycloakUsersState &
  SeamlessImmutable.ImmutableObject<KeycloakUsersState>;

/** initial keycloak users state */
export const initialState: ImmutableKeycloakUsersState = SeamlessImmutable({
  usersById: {},
});

/**
 * the users reducer function
 *
 * @param {object} state - keycloak users states
 * @param {object} action - keycloak users actions
 * @returns {object} - the updated states
 */
export function reducer(
  state: ImmutableKeycloakUsersState = initialState,
  action: KeycloakUsersActionTypes
): ImmutableKeycloakUsersState {
  switch (action.type) {
    case KEYCLOAK_USERS_FETCHED:
      return SeamlessImmutable({
        ...state,
        usersById: { ...state.usersById, ...action.usersById },
      });
    case REMOVE_KEYCLOAK_USERS:
      return SeamlessImmutable({
        ...state,
        usersById: action.usersById,
      });

    default:
      return state;
  }
}

// Selectors

export interface KeycloakUsersFilters {
  id: string[] /** get all users whose ids appear in this array */;
  username?: string /** get users whose username includes text in the username field */;
}

// NON MEMOIZED SELECTORS

/**
 * Gets id from KeycloakUsersFilters
 *
 * @param {any} _ - the redux store
 * @param {Array} props - the keycloak users filters object
 * @returns {string} - the keycloak user id
 */
export const getUserIds = (_: Partial<Store>, props: KeycloakUsersFilters): string[] => props.id;

/**
 * Gets name from KeycloakUsersFilters
 *
 * @param {any} _ - the redux store
 * @param {object} props - the users filters object
 * @returns {string} - the keycloak user name
 */
export const getUsername = (_: Partial<Store>, props: KeycloakUsersFilters): string | undefined =>
  props.username;

/** returns all users in the store as values whose keys are their respective ids
 *
 * @param {any} state - the redux store
 * @returns {object} - users object as values, respective ids as keys
 */
export function getKeycloakUsersById(state: Partial<Store>): { [key: string]: KeycloakUser } {
  return (state as Dictionary)[reducerName].usersById;
}

/** gets keycloak users as an array of user objects
 *
 * @param {any} state - the redux store
 * @returns {Array} - an array of users objs
 */
export function getKeycloakUsersArray(state: Partial<Store>): KeycloakUser[] {
  return values(getKeycloakUsersById(state));
}

/**
 * Gets all practitioners whose name includes phrase given in name filter prop
 *
 * @returns {Array} - practitioners whose name includes phrase given in name filter prop
 */
export const getUsersByUsername = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (res1: KeycloakUser[], res2: string | undefined) => KeycloakUser[]
> =>
  createSelector(getKeycloakUsersArray, getUsername, (usersArray, username) =>
    username
      ? usersArray.filter((user: KeycloakUser) =>
          user.username.toLowerCase().includes(username.toLowerCase())
        )
      : usersArray
  );

// MEMOIZED SELECTORS

/**
 * Gets all users whose identifiers appear in ids filter prop value
 *
 * @returns {Array} - users whose identifiers appear in ids filter prop value
 */
export const getKeycloakUsersByIds = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (
    res1: {
      [key: string]: KeycloakUser;
    },
    res2: string[],
    res3: KeycloakUser[]
  ) => KeycloakUser[]
> =>
  createSelector(
    getKeycloakUsersById,
    getUserIds,
    getKeycloakUsersArray,
    (usersById, ids, usersArray) => {
      if (ids === undefined) {
        return usersArray;
      }
      if (ids.length > 0) {
        return ids.map((id: string) => usersById[id]);
      }
      return [];
    }
  );

/** practitioner array selector factory
 * aggregates response from all applied filters and returns results
 *
 * @returns {Array} - aggregates response from all applied filters and returns results
 */
export const makeKeycloakUsersSelector = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (res1: KeycloakUser[], res2: KeycloakUser[]) => KeycloakUser[]
> =>
  createSelector(getKeycloakUsersByIds(), getUsersByUsername(), (arr1, arr2) => {
    return intersect([arr1, arr2], JSON.stringify);
  });
