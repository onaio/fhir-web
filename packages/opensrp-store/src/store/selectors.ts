import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';
import { Dictionary } from '@onaio/utils';
import { createSelector } from 'reselect';

/** get API Token from the Redux store
 *
 * @param {any} state - the redux store
 */
interface StoreState {
  session: {
    [key: string]: Dictionary;
  };
}

interface APIFilters {
  accessToken?: boolean;
  apiToken?: boolean;
  providerState?: boolean;
}

/** Fetch access token from store if access_token filter val is true
 *
 * @param {object} _ - redux store
 * @param {object} props - api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.accessToken || null;

/** Fetch api token from store if api_token filter val is true
 *
 * @param {object} _ redux store
 * @param {object} props api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchApiToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.apiToken || null;

/** Fetch oauth provider state from store if providerState filter val is true
 *
 * @param {Object} _ redux store
 * @param {Object} props api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchOauthProviderState = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.providerState || null;

/** Gets extra data object from store
 *
 * @param {object} state - redux store
 * @returns {Object}
 */

export const getExtraData = (state: Partial<Store>): Dictionary =>
  (state as StoreState)[sessionReducerName].extraData;

/** API state selector
 *
 * @returns {Object} - the states
 */
export const makeAPIStateSelector = () =>
  createSelector(
    [fetchAccessToken, fetchApiToken, fetchOauthProviderState, getExtraData],
    (accessToken, apiToken, oauthState, extraData) => {
      if (accessToken) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.access_token) || null;
      } else if (apiToken) {
        return extraData.api_token || null;
      } else if (oauthState) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.state) || null;
      }
    }
  );
