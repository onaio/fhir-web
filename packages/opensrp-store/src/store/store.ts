import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';
import { Dictionary } from '@onaio/utils';

/** declare globals interface */
declare global {
  interface Window {
    __PRELOADED_STATE__: Dictionary;
  }
}

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__ || {};
// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__;

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
export const store = getConnectedStore(defaultReducers, preloadedState);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});
