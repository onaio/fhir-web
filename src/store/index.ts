import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__ || {};
// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__;

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers, preloadedState);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});

export default store;
