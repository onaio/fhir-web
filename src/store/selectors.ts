import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';

/** get API Token from the Redux store
 * @param {Partial<Store>} state - the redux store
 */
export function getApiToken(state: Partial<Store>): string {
  const extraData = (state as any)[sessionReducerName].extraData;
  return extraData.api_token || null;
}

/** get Access Token from the Redux store
 * @param {Partial<Store>} state - the redux store
 */
export function getAccessToken(state: Partial<Store>): string | null {
  const extraData = (state as any)[sessionReducerName].extraData;
  if (extraData.oAuth2Data && extraData.oAuth2Data.access_token) {
    return extraData.oAuth2Data.access_token;
  }
  return null;
}

/** get the oAuth2 provider state parameter from the Redux store
 * @param {Partial<Store>} state - the redux store
 */
export function getOauthProviderState(state: Partial<Store>): string | null {
  const extraData = (state as any)[sessionReducerName].extraData;
  if (extraData.oAuth2Data && extraData.oAuth2Data.state) {
    return extraData.oAuth2Data.state;
  }
  return null;
}
