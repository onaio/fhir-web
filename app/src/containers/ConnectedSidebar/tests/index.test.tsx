import { authenticateUser } from '@onaio/session-reducer';
import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import ConnectedSidebar from '..';
import { store } from '@opensrp/store';

jest.mock('../../../configs/env');

describe('components/ConnectedSidebar', () => {
  it('renders the ConnectedSidebar component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('aside').props()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders user managment menu for users with appropriate role', () => {
    store.dispatch(
      authenticateUser(
        true,
        {
          email: 'bob@example.com',
          name: 'Bobbie',
          username: 'RobertBaratheon',
        },
        {
          roles: ['ROLE_EDIT_KEYCLOAK_USERS'],
          username: 'superset-user',
          user_id: 'cab07278-c77b-4bc7-b154-bcbf01b7d35b',
        }
      )
    );
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });

  it('displays menu links for enabled modules', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_FORM_CONFIGURATION = 'true';
    envModule.ENABLE_PRODUCT_CATALOGUE = 'true';
    envModule.ENABLE_PLANS = 'true';
    envModule.ENABLE_CARD_SUPPORT = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });

  it('displays menu links for enabled Location module', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_LOCATIONS = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });

  it('correctly expand users menu', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{ pathname: `/admin`, hash: '', search: '', state: {} }]}>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });

  it('correctly expand location menu', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_LOCATIONS = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{ pathname: `/location`, hash: '', search: '', state: {} }]}>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });

  it('correctly expand admin-form-config menu', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_LOCATIONS = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{ pathname: `/draft`, hash: '', search: '', state: {} }]}>
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });

  it('correctly expand teams menu', () => {
    const envModule = require('../../../configs/env');
    envModule.ENABLE_TEAMS = 'true';

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter
          initialEntries={[{ pathname: `/admin/teams`, hash: '', search: '', state: {} }]}
        >
          <ConnectedSidebar />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('Menu').at(0).prop('children')).toMatchSnapshot();
    wrapper.unmount();
  });
});
