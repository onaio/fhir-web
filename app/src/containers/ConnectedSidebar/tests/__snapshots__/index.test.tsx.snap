// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ConnectedSidebar correctly expand admin-form-config menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configuration"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand location menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configuration"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar correctly expand users menu 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configuration"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar displays menu links for enabled Location module 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Locations"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/unit"
        >
          Locations unit
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/location/group"
        >
          Locations unit group
        </Link>
      </MenuItem>
    </SubMenu>
    <SubMenu
      title="Form Configuration"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar displays menu links for enabled modules 1`] = `
Array [
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Missions"
  >
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/active"
      >
        Active
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/draft"
      >
        Draft
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/complete"
      >
        Complete
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/missions/trash"
      >
        Trash
      </Link>
    </MenuItem>
  </SubMenu>,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
    <SubMenu
      title="Form Configuration"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/releases"
        >
          Manifest Releases
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/drafts"
        >
          Draft Files
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/form-config/json-validators"
        >
          JSON Validators
        </Link>
      </MenuItem>
    </SubMenu>
  </SubMenu>,
]
`;

exports[`components/ConnectedSidebar renders the ConnectedSidebar component 1`] = `
Object {
  "children": Array [
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      >
        <Link
          to="/"
        >
          <img
            alt=""
            className="img-fluid"
            src="test-file-stub"
          />
        </Link>
      </div>
      <Menu
        className="menu-dark"
        defaultOpenKeys={
          Array [
            "",
          ]
        }
        defaultSelectedKeys={
          Array [
            "",
          ]
        }
        mode="inline"
        selectedKeys={
          Array [
            "",
          ]
        }
        theme="dark"
      >
        <SubMenu
          icon={<ForwardRef(DashboardOutlined) />}
          title="Admin"
        >
          <MenuItem>
            <Link
              className="admin-link"
              to="/admin/teams"
            >
              Teams
            </Link>
          </MenuItem>
          <MenuItem>
            <Link
              className="admin-link"
              to="/product-catalogue/list"
            >
              Product Catalogue
            </Link>
          </MenuItem>
        </SubMenu>
      </Menu>
    </div>,
    null,
  ],
  "className": "ant-layout-sider ant-layout-sider-dark layout-sider",
  "style": Object {
    "flex": "0 0 275px",
    "maxWidth": "275px",
    "minWidth": "275px",
    "width": "275px",
  },
}
`;

exports[`components/ConnectedSidebar renders user managment menu for users with appropriate role 1`] = `
Array [
  undefined,
  <SubMenu
    icon={<ForwardRef(DashboardOutlined) />}
    title="Admin"
  >
    <SubMenu
      title="Users"
    >
      <MenuItem>
        <Link
          className="admin-link"
          to="/admin/users/list"
        >
          User Management
        </Link>
      </MenuItem>
    </SubMenu>
    <MenuItem>
      <Link
        className="admin-link"
        to="/admin/teams"
      >
        Teams
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        className="admin-link"
        to="/product-catalogue/list"
      >
        Product Catalogue
      </Link>
    </MenuItem>
  </SubMenu>,
]
`;
